src/snowflake/connector/bind_upload_agent.py

```diff
@@ -38,13 +39,29 @@ def __init__(
             rows: Rows of binding parameters in CSV format.
             stream_buffer_size: Size of each file, default to 10MB.
         """
+        self._use_scoped_temp_object = (
+            cursor.connection._session_parameters.get(
+                _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, False
+            )
+            if cursor.connection._session_parameters
```

Comment:
(not a strong opinion) Will `cursor.connection._session_parameters` ever be None? iirc we always at least initialized to an empty dict, so we can even simplify do 
```python
self._use_scoped_temp_object = cursor.connection._session_parameters.get(
    _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, False
)
```
But to err on the side of caution is also a good idea.

Reply:
I followed the practice in snowpark here, and this approach handled all possible errors too

Reply:
We hit an issue with this change. The latest `3.12.3` broke our tests.

In our tests, we have a `MockConnection` class which mocks all the public functions of Connection. This change, however, started using a private class member (`connection._session_parameters`), which doesn't seem right.